route/build/src/kwt_route.f90

```diff
@@ -478,6 +482,18 @@ subroutine QROUTE_RCH(IENS,JRCH,    & ! input: array indices
     else
       ! set flow in headwater reaches to modelled streamflow from time delay histogram
       RCHFLX(IENS,JRCH)%REACH_Q = RCHFLX(IENS,JRCH)%BASIN_QR(1) 
+      if (allocated(KROUTE(IENS,JRCH)%KWAVE)) THEN
```

Comment:
if there is a way to deallocate naturally at the end of a use of a variable, rather than have to check before allocating, I think that would be preferable.  in some cases that may be difficult to track, though.  also, if the reallocated variable has the same size as in its former use then just reusing it without de/re allocation will be more efficient.  I don't know if that's the case here.